src/terrain/clients/iplant_groups.clj

```diff
@@ -166,3 +174,49 @@
         group  (format "%s:%s" folder name)]
     (verify-group-exists client user group)
     (c/remove-group-members client user group members)))
+
+;; Team Functions
+
+(def ^:private team-group-type "group")
+
+(defn- get-team-folder-name [client & [user]]
+  (if-not user
+    (c/build-folder-name client "teams")
+    (c/build-folder-name client (format "teams:%s" user))))
+
+(defn- ensure-team-folder-exists [client user]
+  (ensure-folder-exists client (config/grouper-user) (get-team-folder-name client))
+  (ensure-folder-exists client user (get-team-folder-name client user)))
+
+(defn- get-teams* [client user search-folder lookup-fn]
+  (let [folder (get-team-folder-name client)]
+    (get-groups* search-folder (partial format-group folder) client user lookup-fn)))
+
+(defn- filter-teams [search result]
+  (update-in result [:groups] (partial filter (comp (partial re-find (re-pattern (str "\\Q" search))) :name))))
```

Comment:
This is intended to approximate Grouper's `FIND_BY_GROUP_NAME_APPROXIMATE` search option, which isn't available when searching for groups that a subject belongs to. There's not much documentation on this search option, so my approximate group name search approximation may be off a little. (Say _that_ five times fast. 😛 )